File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int left = 0;
        Set<Integer> hs = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            if ((i - left) > k) {
                hs.remove(nums[left]);
                left++;
            }

            if (hs.contains(nums[i])) return true;
            hs.add(nums[i]);
        }
       
        return false;
    }
}

File: Data Structures & Algorithms/largest-unique-number/submission-0.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        s = set()
        tracked = set()

        for n in nums:
            if n in s:
                s.remove(n)
                tracked.add(n)

            if n not in tracked:
                s.add(n)
        
        return max(s or [-1])